src/main/java/org/truffleruby/RubyContext.java

```diff
@@ -412,52 +410,17 @@ private String findRubyHome() throws IOException {
             return home.getCanonicalPath();
         }
 
-        // Try to find it automatically from the location of the JAR, but this won't work from the JRuby launcher as it uses the boot classpath
-
-        if (TruffleOptions.AOT) {
-            final String executablePath = (String) Compiler.command(new Object[]{"com.oracle.svm.core.posix.GetExecutableName"});
-            final String parentDirectory = new File(executablePath).getParent();
-
-            // Root of the GraalVM distribution.
-            File candidate = Paths.get(parentDirectory, "language", "ruby").toFile();
+        if (!options.LAUNCHER.isEmpty()) {
+            final File candidate = Paths.get(options.LAUNCHER).getParent().getParent().toFile();
             if (isRubyHome(candidate)) {
                 return candidate.getCanonicalPath();
             }
-
-            // Root of the TruffleRuby source tree.
-            if (isRubyHome(new File(parentDirectory))) {
-                return new File(parentDirectory).getCanonicalPath();
-            }
-
-            // Nested mx build.
-            candidate = Paths.get(parentDirectory, "..", "..", "..", "truffleruby").toFile();
-            if (isRubyHome(candidate)) {
-                return candidate.getCanonicalPath();
-            }
-        } else {
-            final CodeSource codeSource = getClass().getProtectionDomain().getCodeSource();
-
-            if (codeSource != null && codeSource.getLocation().getProtocol().equals("file")) {
-                final File jar = new File(codeSource.getLocation().getFile()).getCanonicalFile();
-                final File jarDir = jar.getParentFile();
-
-                if (jarDir.toPath().endsWith(Paths.get("mxbuild", "dists"))) {
-                    final File candidate = jarDir.getParentFile().getParentFile();
-                    if (isRubyHome(candidate)) {
-                        // TruffleRuby Source build
-                        return candidate.getCanonicalPath();
-                    }
-                }
-
-                if (jarDir.getName().equals("ruby") && isRubyHome(jarDir)) {
-                    // GraalVM build or distribution
-                    return jarDir.getCanonicalPath();
-                }
-            }
```

Comment:
How is the home found if Ruby is invoked from another language, so that neither -Xhome nor -Xlauncher is set?
This should continue to work, at least in GraalVM.

Reply:
I think the simplification of getRubyHome() does not belong to this PR. It should be fixed for AOT separately (and there it's not clear whether the image is in bin/). For JVM I think the current logic is already rather simple.

Reply:
It's assumed that an image is placed in standard Ruby dir structure or the home is specified explicitly. The graalvm.home property is now used to determine the ruby home as well, c5fbeda.